src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1857 @@
+                sd_bus_message *msg,
+                const Job *j,
+                sd_json_variant *json,
+                sd_bus_error *error) {
+
+        uint64_t instances;
+        
+        assert(json);
+
+        instances = sd_json_variant_unsigned(sd_json_variant_by_key(json, "removed"));
+        
+        return sd_bus_reply_method_return(msg, "u", instances);
+}
+
+static int target_method_vacuum(sd_bus_message *msg, void *userdata, sd_bus_error *error) {
+        Target *t = ASSERT_PTR(userdata);
+        _cleanup_(job_freep) Job *j = NULL;
+        int r;
+
+        assert(msg);
+
+        const char *details[] = {
+                "class", target_class_to_string(t->class),
+                "name", t->name,
+                NULL
+        };
+
+        r = bus_verify_polkit_async(
+                msg,
+                "org.freedesktop.sysupdate1.vacuum",
+                details,
+                &t->manager->polkit_registry,
+                error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1; /* Will call us back */
+
+        r = job_new(JOB_VACUUM, t, msg, target_method_vacuum_finish, &j);
+        if (r < 0)
+                return r;
+
+        r = job_start(j);
+        if (r < 0)
+                return sd_bus_error_set_errnof(error, r, "Failed to start job: %m");
+        TAKE_PTR(j); /* Avoid job from being killed & freed */
+
+        return 1;
+}
+
+static int target_method_get_version(sd_bus_message *msg, void *userdata, sd_bus_error *error) {
+        Target *t = ASSERT_PTR(userdata);
+        _cleanup_free_ char *target_arg = NULL;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        int r;
+
+        r = target_get_argument(t, &target_arg);
+        if (r < 0)
+                return r;
+
+        r = sysupdate_run_simple(&v, "--offline", "list", target_arg, NULL);
+        if (r < 0)
+                return r;
+
+        sd_json_variant *version_json = sd_json_variant_by_key(v, "current");
+        if (!version_json)
+                return -EINVAL;
+
+        if (sd_json_variant_is_null(version_json))
+                return sd_bus_reply_method_return(msg, "s", "");
+
+        if (!sd_json_variant_is_string(version_json))
+                return -EINVAL;
+
+        return sd_bus_reply_method_return(msg, "s", sd_json_variant_string(version_json));
+}
+
+static int target_get_appstream(char ***ret, Target *t) {
+        _cleanup_free_ char *target_arg = NULL;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        int r;
+
+        r = target_get_argument(t, &target_arg);
+        if (r < 0)
+                return r;
+
+        r = sysupdate_run_simple(&v, "--offline", "list", target_arg, NULL);
+        if (r < 0)
+                return r;
+
+        sd_json_variant *appstream_url_json = sd_json_variant_by_key(v, "appstream_urls");
+        if (!appstream_url_json)
+                return -EINVAL;
```

Comment:
and here too (please always add debug logging when we parse stuff and it doesn't match our expectations of formatting)